# Problemas de prueba de hipótesis

## Ejercicio 1

Usted es un nutricionista que investiga dos tipos diferentes de dietas para ver si existe una diferencia significativa en la pérdida de peso después de un mes. Eliges dos grupos aleatorios de personas; un grupo sigue la primera dieta y el otro sigue la segunda. Al final del mes, se registra la pérdida de peso (en kg) de cada persona:

| Diet 1 | Diet 2 |
|:-------|:-------|
| 2.0 | 3.0 |
| 2.5 | 3.2 |
| 3.0 | 3.1 |
| 2.8 | 2.9 |
| 2.3 | 2.8 |
| 2.7 | 3.0 |
| 2.5 | 3.2 |

Con estos datos se busca responder a la siguiente pregunta: ¿Existe una diferencia significativa en la pérdida de peso promedio entre las personas que siguieron la primera dieta y las que siguieron la segunda dieta?

Para sacar conclusiones, siga los siguientes puntos:

-Enunciar la hipótesis: hipótesis nula y alternativa.
-Realizar la prueba para comprobar la hipótesis. Puede utilizar la prueba t de Student.
-Analizar las conclusiones.

In [ ]:
import numpy as np
import pandas as pd
from scipy.stats import ttest_ind
from scipy.stats import f_oneway

In [3]:
diet_1 = np.array([2.0, 2.5, 3.0, 2.8, 2.3, 2.7, 2.5])
diet_2 = np.array([3.0, 3.2, 3.1, 2.9, 2.8, 3.0, 3.2])

alpha = 0.05

print("H0: no existe diferencia significativa entre las medias de perdida de peso de ambas dietas.")
print("H1: si existe diferencia significativa entre las medias de perdida de peso de ambas dietas.\n")

t_stat, p_value = ttest_ind(diet_1, diet_2)

results = pd.DataFrame({
    "grupo": ["Diet 1", "Diet 2"],
    "media": [diet_1.mean(), diet_2.mean()],
    "desviacion_std": [diet_1.std(ddof=1), diet_2.std(ddof=1)],
    "n": [len(diet_1), len(diet_2)]
})

display(results)

print(f"t-statistic: {t_stat:.4f}")
print(f"p-value: {p_value:.4f}")
print(f"alpha: {alpha}")

if p_value < alpha:
    print("\nDecision: rechazo H0")
else:
    print("\nDecision: no rechazo H0")

H0: no existe diferencia significativa entre las medias de perdida de peso de ambas dietas.
H1: si existe diferencia significativa entre las medias de perdida de peso de ambas dietas.



,grupo,media,desviacion_std,n
0,Diet 1,2.542857,0.330944,7
1,Diet 2,3.028571,0.149603,7


t-statistic: -3.5383
p-value: 0.0041
alpha: 0.05

Decision: rechazo H0


<div style="
  background: #ececec;
  border-left: 6px solid #22c55e;
  border-radius: 12px;
  padding: 16px 18px;
  box-shadow: 0 8px 24px rgba(15,23,42,.08);
  margin: 6px 0 2px 0;
">
  <p style="margin: 0; line-height: 1.75; color: #111827; font-size: 16px;">
    Veo que la media de <span style="color:#dc2626; font-weight:700;">Diet 2</span> es mayor que la de <span style="color:#dc2626; font-weight:700;">Diet 1</span>. Como el <span style="color:#dc2626; font-weight:700;">p-value = 0.0041</span> es menor que <span style="color:#dc2626; font-weight:700;">0.05</span>, rechazo <span style="color:#dc2626; font-weight:700;">H0</span>. Esto me dice que si hay una diferencia significativa entre ambas dietas. En estos datos, <span style="color:#dc2626; font-weight:700;">Diet 2</span> muestra una perdida de peso promedio mas alta.
  </p>
</div>

## ANOVA

**ANOVA** (*Analysis of Variance - Análisis de variación*) Es una técnica estadística utilizada para comparar las medidas de dos o más grupos. La idea detrás de ANOVA es descomponer la variabilidad total de los datos en dos componentes: variabilidad entre grupos y variabilidad dentro de los grupos:

- **Variabilidad entre grupos**: Esta variabilidad se refiere a las diferencias entre las medias del grupo. Si esta variabilidad es considerablemente mayor que la variabilidad dentro del grupo, podría ser una indicación de que al menos una de las medias del grupo es diferente.
- **Variabilidad dentro del grupo**: Esta variabilidad se refiere a la dispersión de los datos dentro de cada grupo. Si todos los grupos tienen una variabilidad similar, entonces cualquier diferencia notable en las medias de los grupos podría considerarse significativa.

Las hipótesis en ANOVA suelen incluir:

- **Hipótesis nula** ($H₀$): Las medias de todos los grupos son iguales.
- **Hipótesis alternativa** ($H₁$): Al menos una de las medias del grupo es diferente.

Si el resultado de la prueba ANOVA es significativo (por ejemplo, un valor p inferior a un umbral como 0,05), esto sugiere que al menos la media de un grupo es diferente.

## Ejercicio 2

Un agricultor decide probar tres tipos diferentes de fertilizantes para determinar si alguno es superior en términos de producción de maíz. El agricultor planta maíz en 15 parcelas idénticas y utiliza los tres fertilizantes (5 parcelas para cada tipo). Al final de la temporada mide el rendimiento de maíz (en kg) de cada parcela, con el siguiente resultado:

| Fertilizante 1 | Fertilizante 2 | Fertilizante 3 |
|:-------------|:-------------|:-------------|
| 20 | 22 | 24 |
| 21 | 21 | 23 |
| 20 | 23 | 22 |
| 19 | 22 | 23 |
| 20 | 21 | 24 |

Con estos datos busca responder la siguiente pregunta: ¿Existe una diferencia significativa en el rendimiento promedio del maíz entre los tres tipos de fertilizantes?

Para ayudarte, sigue los siguientes puntos:

-Enunciar la hipótesis: hipótesis nula y alternativa.
-Realizar la prueba ANOVA.
-Analizar las conclusiones.
-Si un fertilizante es mejor que otro, ¿cómo podemos saberlo?

In [5]:
fertilizante_1 = np.array([20, 21, 20, 19, 20])
fertilizante_2 = np.array([22, 21, 23, 22, 21])
fertilizante_3 = np.array([24, 23, 22, 23, 24])

alpha = 0.05

print("H0: las medias de rendimiento de los tres fertilizantes son iguales.")
print("H1: al menos una media de rendimiento es diferente.\n")

f_stat, p_value = f_oneway(fertilizante_1, fertilizante_2, fertilizante_3)

results = pd.DataFrame({
    "grupo": ["Fertilizante 1", "Fertilizante 2", "Fertilizante 3"],
    "media": [fertilizante_1.mean(), fertilizante_2.mean(), fertilizante_3.mean()],
    "desviacion_std": [
        fertilizante_1.std(ddof=1),
        fertilizante_2.std(ddof=1),
        fertilizante_3.std(ddof=1)
    ],
    "n": [len(fertilizante_1), len(fertilizante_2), len(fertilizante_3)]
})

display(results)

print(f"F-statistic: {f_stat:.4f}")
print(f"p-value: {p_value:.6f}")
print(f"alpha: {alpha}")

if p_value < alpha:
    print("\nDecision: rechazo H0")
else:
    print("\nDecision: no rechazo H0")

H0: las medias de rendimiento de los tres fertilizantes son iguales.
H1: al menos una media de rendimiento es diferente.



,grupo,media,desviacion_std,n
0,Fertilizante 1,20.0,0.707107,5
1,Fertilizante 2,21.8,0.836660,5
2,Fertilizante 3,23.2,0.836660,5


F-statistic: 20.3158
p-value: 0.000140
alpha: 0.05

Decision: rechazo H0


<div style="
  background: #ececec;
  border-left: 6px solid #22c55e;
  border-radius: 12px;
  padding: 16px 18px;
  box-shadow: 0 8px 24px rgba(15,23,42,.08);
  margin: 6px 0 2px 0;
">
  <p style="margin: 0; line-height: 1.75; color: #111827; font-size: 16px;">
    Veo que las medias cambian entre los tres grupos. <span style="color:#dc2626; font-weight:700;">Fertilizante 1</span> tiene una media de <span style="color:#dc2626; font-weight:700;">20.0</span>, <span style="color:#dc2626; font-weight:700;">Fertilizante 2</span> llega a <span style="color:#dc2626; font-weight:700;">21.8</span> y <span style="color:#dc2626; font-weight:700;">Fertilizante 3</span> alcanza <span style="color:#dc2626; font-weight:700;">23.2</span>. Como el <span style="color:#dc2626; font-weight:700;">p-value = 0.000140</span> es menor que <span style="color:#dc2626; font-weight:700;">0.05</span>, rechazo <span style="color:#dc2626; font-weight:700;">H0</span>. Esto me dice que si hay diferencias significativas en el rendimiento promedio. Con este resultado puedo sospechar que <span style="color:#dc2626; font-weight:700;">Fertilizante 3</span> rinde mejor, pero para confirmar exactamente entre que grupos esta la diferencia tendria que hacer una prueba <span style="color:#dc2626; font-weight:700;">post hoc</span>, por ejemplo <span style="color:#dc2626; font-weight:700;">Tukey</span>.
  </p>
</div>